React/Views/RCTView.m

```diff
@@ -184,6 +184,54 @@ - (BOOL)didActivateAccessibilityCustomAction:(UIAccessibilityCustomAction *)acti
   return YES;
 }
 
+- (NSString *)accessibilityValue
+{
+  NSMutableString *value = [NSMutableString stringWithString:@""];
+  NSDictionary *roleDescriptions = @{
+                                     @"alert" : @"alert",
+                                     @"checkbox" : @"checkbox",
+                                     @"combobox" : @"combo box",
+                                     @"editabletext" : @"edit",
+                                     @"menu" : @"menu",
+                                     @"menubar" : @"menu bar",
+                                     @"menuitem" : @"menu item",
+                                     @"progressbar" : @"progress bar",
+                                     @"radiobutton" : @"radio button",
+                                     @"radiogroup" : @"radio group",
+                                     @"scrollbar" : @"scroll bar",
+                                     @"spinbutton" : @"spin button",
+                                     @"switch" : @"switch",
```

Comment:
First of all, thanks for working on it.

1) What do you think of using `UISwitch().accessibilityTraits` for it? It would bring the default and **localizable** trait for switch role.

I was able to make it work (in my local react-native fork), based on this example https://github.com/jldailey13/24-Accessibility-Article-Examples/blob/master/24A11y%20Article/Name%20Examples/AccessibleSwitchView.swift

2) Swith component will benefit from this role https://github.com/facebook/react-native/blob/8774229f668adec94c1b76cc53e260f4a9354797/Libraries/Components/Switch/Switch.js#L169 👍

Reply:
Interestingly, this does seem correct, but we can't get it to work in our local trees. Switches are still spoken as buttons with a value of 1 or 0. Not sure what we're doing wrong...

Reply:
Hey @jldailey13, take a look @marcmulcahy has an interesting issue here!

Reply:
@marcmulcahy

I was able to get the right trait by instantiating a new `UISwitch`

```
@synthesize a11ySwitch = _a11ySwitch;

....

_a11ySwitch = [UISwitch new];
view.reactAccessibilityElement.accessibilityTraits = [_a11ySwitch accessibilityTraits];
view.reactAccessibilityElement.accessibilityValue = isOn ? @"1" : @"0";

````

Reply:
It looks like UISwitch is simply setting traits to Button. We tried something similar, but VoiceOver doesn't present the switch as a switch-- it says "button 1". Did you spin this up with VoiceOver and confirm that it properly describes the switch?

Reply:
Weird. This is the full commit https://github.com/elucaswork/react-native/commit/80a304386101829baede42f2af7bf5813348dbb2. 
and yes, I was testing it with VoiceOver turned on. (I was very excited with the result)

Reply:
That is exciting! I'll look closer at this full commit tomorrow, we must have missed something. It may be that VoiceOver is looking at more than the traits, and the class of the actual object is significant?

Reply:
If you are making the accessibilityTraits of `RCTView` equal to `UISwitch().accessibilityTraits`, this should work, assuming that `RCTView` is inheriting from `UIView`.  I'm not sure what would be causing this to fail, given the information.

Reply:
I looked a bit closer at this. @elucaswork, how exactly did you test? I know you used VoiceOver-- which components did you play with?

The React Native switch component already behaved itself, since it uses a UISwitch underneath, so if that's what you used to test, you wouldn't see any change. It used to work before this change, and still does.

To validate the actual role, we created generic views (UIView underneath) and set the role to "switch". after applying what I think is a change similar to yours, VoiceOver says "label 1 button", which isn't what we want.

What I fear is going on is that VoiceOver is special case handling components based on their class names, not just their accessibility traits.


I hope I'm wrong, and we just missed a little detail. Let me know.

Reply:
I've tested it with a `TouchableOpacity` (UIView as well):

```
<TouchableOpacity
  onPress={() => this.setState({switchOn: !this.state.switchOn})}
  accessibilityRole="switch"
  accessibilityStates={this.state.switchOn ? ['on'] : ['off']}>
  <Text>
    The switch value is {this.state.switchOn ? 'on' : 'off'}
  </Text>
</TouchableOpacity>
```

I'm gonna try later to play with this PR and see if I can replicate the same result that I got on my local branch. I let you know.

Reply:
Interesting. Good to know. We can test with your patch here too... Although it conflicts with this PR, so we'll need to do a bit of surgery. :)

Reply:
There's actually a non-public trait that UISwitches use and it's `UIAccessibilityTraitToggle`. It's defined as `0x0020000000000000` ( 1 << 53 ). 

So you could do something like `switch.accessibilityTraits = UIAccessibilityTraitButton | 0x0020000000000000` (or define a constant) to get this behavior

Reply:
As an aside, while this constant isn't in the header, it's in the source so you can print out this symbol in lldb:

```
(lldb) po UIAccessibilityTraitToggle
0x0020000000000000
```